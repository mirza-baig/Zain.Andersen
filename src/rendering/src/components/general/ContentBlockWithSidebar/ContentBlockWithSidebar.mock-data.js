const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'ContentBlockWithSidebar',
  },
  fields: {
    "componentSpacing": null,
    "sectionId": {
      "value": ""
    },
    "headlineLevel": {
      "id": "7f10288a-beb1-4d95-a139-32df40094441",
      "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
      "name": "Heading 3",
      "displayName": "Heading 3",
      "fields": {
        "Value": {
          "value": "h2"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "headlineText": {
      "value": "Headline text"
    },
    "body": {
      "value": "Test Body"
    },
    "subheadlineLevel": null,
    "subheadlineText": {
      "value": "Test SubHeadline"
    },
    "sidebar": {
      "value": "Side copy"
    },
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
