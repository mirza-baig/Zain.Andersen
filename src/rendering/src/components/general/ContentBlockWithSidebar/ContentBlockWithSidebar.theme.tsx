// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const ContentBlockWithSidebarTheme: ThemeFile = {
  aw: {
    classes: {
      /** Insert Theme classes here **/
      leftColumnClass: 'grid-cols-12 col-span-9',
      rightColumnClass:
        'sm-px-0 border-t col-span-9 border-gray md:col-span-3 md:p-3 md:border-l md:border-t-0',
      headlineClass: 'text-theme-text text-sm-m md:text-m font-semibold mb-s leading-6}',
      bodyClass: 'text-theme-body text-body md:pr-3 lg:pr-[120px] mb-s',
      subBodyClass: 'text-small',
      subheadlineClass:
        'text-theme-text font-sans text-sm-s md:text-s mt-s md:mt-0 mb-xxs font-semibold',
    },
  },
  rba: {
    classes: {
      /** Insert Theme classes here **/
      leftColumnClass: 'grid-cols-12 col-span-9',
      rightColumnClass:
        'sm-px-0 border-t col-span-9 border-gray md:col-span-3 md:p-3 md:border-l md:border-t-0',
      headline: 'text-theme-text text-sm-m md:text-m font-bold mb-s',
      bodyClass: 'text-theme-body text-body md:pr-3 lg:pr-[120px] mb-s',
      subBodyClass: 'text-theme-body text-body mb-s',
      subheadlineClass:
        'text-theme-text font-sans text-sm-s md:text-s mt-s md:mt-0 mb-xxs font-bold',
    },
  },
};
