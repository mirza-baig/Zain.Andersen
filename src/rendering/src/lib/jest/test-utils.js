// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
