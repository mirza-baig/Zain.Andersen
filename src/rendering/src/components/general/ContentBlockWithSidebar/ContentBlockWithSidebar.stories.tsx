// Global
import { Story, Meta } from '@storybook/react';
// Local
import ContentBlockWithSidebar, { ContentBlockWithSidebarProps } from './ContentBlockWithSidebar';
import defaultData from './ContentBlockWithSidebar.mock-data';

export default {
  title: 'General/ContentBlockWithSidebar',
  component: ContentBlockWithSidebar,
} as Meta;
const Template: Story<ContentBlockWithSidebarProps> = (props) => (
  <ContentBlockWithSidebar {...props} />
);
export const Default = Template.bind({});
Default.args = defaultData;
