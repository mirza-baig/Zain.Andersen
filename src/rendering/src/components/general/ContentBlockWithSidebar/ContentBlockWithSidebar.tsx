// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck, Text } from '@sitecore-jss/sitecore-jss-nextjs';
import RichTextWrapper from 'src/helpers/RichTextWrapper/RichTextWrapper';

// Components
import { Component } from 'src/helpers/Component';
import { ContentBlockWithSidebarTheme } from './ContentBlockWithSidebar.theme';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { Headline } from 'src/helpers/Headline';

export type ContentBlockWithSidebarProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.General.ContentBlockWithSidebar.ContentBlockWithSidebar;
const ContentBlockWithSidebar = (props: ContentBlockWithSidebarProps) => {
  const { themeData } = useTheme(ContentBlockWithSidebarTheme);

  return (
    <Component variant="lg" gap=" " dataComponent="general/contentblockwithsidebar" {...props}>
      <div className={themeData.classes.leftColumnClass}>
        <Headline defaultTag="h2" classes={themeData.classes.headlineClass} {...props} />
        <BodyCopy {...props} />
      </div>

      <div className={themeData.classes.rightColumnClass}>
        <Text
          tag={'h3'}
          className={themeData.classes.subheadlineClass}
          field={props.fields?.sideBarSubheading}
        />
        <div>
          <RichTextWrapper
            classes={themeData.classes.subBodyClass}
            field={props.fields?.sideBarBodyCopy}
          />
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<ContentBlockWithSidebarProps>(ContentBlockWithSidebar);
