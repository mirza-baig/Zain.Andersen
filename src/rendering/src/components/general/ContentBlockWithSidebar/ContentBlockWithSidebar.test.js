// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import ContentBlockWithSidebar from './ContentBlockWithSidebar';
import defaultData from './ContentBlockWithSidebar.mock-data';

it('renders correctly', async () => {
  const component = snapshot(ContentBlockWithSidebar, { componentProps: defaultData });
  await hasDataComponent(component, 'general/contentblockwithsidebar');
});
