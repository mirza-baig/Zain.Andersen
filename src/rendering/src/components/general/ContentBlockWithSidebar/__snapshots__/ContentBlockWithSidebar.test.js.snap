// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/contentblockwithsidebar"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m   "
    >
      <div
        class="grid-cols-12 col-span-9"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-semibold mb-s leading-6}"
        >
          <h2>
            Headline text
          </h2>
        </div>
        <div
          class="text-theme-body text-body mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Test Body
            </div>
          </div>
        </div>
      </div>
      <div
        class="sm-px-0 border-t col-span-9 border-gray md:col-span-3 md:p-3 md:border-l md:border-t-0"
      >
        <div />
      </div>
    </div>
  </div>
</section>
`;
